### PR TITLE
Partitioned comm: prereq of the main partitioned comm PR

### PR DESCRIPTION
**Pull request by minsii**
## Pull Request Description
Code refactoring and format cleanup with existing code. This is prereq of the main partitioned comm PR.

## Author Checklist
* [ ] Reference appropriate issues (with "Fixes" or "See" as appropriate)
* [ ] Remove xfail from the test suite when fixing a test
* [ ] Commits are self-contained and do not do two things at once
* [ ] Commit message is of the form: `module: short description` and follows [good practice](https://chris.beams.io/posts/git-commit/)
* [ ] Passes whitespace checkers
* [ ] Passes warning tests
* [ ] Passes all tests
* [ ] Add comments such that someone without knowledge of the code could understand
* [ ] You or your company has a signed contributor's agreement on file with Argonne
* [ ] For non-Argonne authors, request an explicit comment from your companies PR approval manager

### CODE REVIEW

**Comment by minsii** [comment]
test:mpich/ch4/most
test:mpich/ch3/most

File: src/mpid/ch4/errnames.txt [hunk 1]
```diff
@@ -6,6 +6,8 @@
 **ch4|badmsgtype:request contained an invalid message type
 **ch4|badreqtype %d:request contained an invalid request type (%d)
 **ch4|badreqtype:request contained an invalid request type
+**ch4|badstartreq %d:start an invalid request kind (%d)
+**ch4|badstartreq:start an invalid request kind
```
Comment (hzhou) [inline src/mpid/ch4/errnames.txt:10]: Do we really need a different message? They read the same to me.

**Review by hzhou — COMMENTED** [review]
Looks good. Just a small comment.

File: src/mpid/ch4/errnames.txt (hunk 1 above)
Comment (minsii) [reply src/mpid/ch4/errnames.txt:10]: Do you mean `**ch4|badreqtype` v.s. `**ch4|badstartreq`? The former checks persistent req internal type `MPIDI_PREQUEST(preq, p_type)`, but the latter checks `req->kind`.

Comment (hzhou) [reply src/mpid/ch4/errnames.txt:10]: I see.

**Review by hzhou — APPROVED** [review]

**Comment by minsii** [comment]
@hzhou The TCP fail on centos64 seems irrelevant to this PR. Are you OK if I go ahead and merge?

**Comment by hzhou** [comment]
> @hzhou The TCP fail on centos64 seems irrelevant to this PR. Are you OK if I go ahead and merge?

Yes, go ahead.

**Pull request merged**